lua/apisix/plugins/proxy-cache.lua

```diff
@@ -186,22 +186,27 @@ function _M.header_filter(conf, ctx)
     core.log.info("proxy-cache plugin header filter phase, conf: ", core.json.delay_encode(conf))
 
     local no_cache = "1"
+    local match_method, match_status = false, false
```

Comment:
We can move this check logic into a single function, which will make the code simpler.